sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -221,4 +221,40 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
       emptyTableData.agg(sumDistinct('a)),
       Row(null))
   }
+
+  test("moments") {
+    checkAnswer(
+      testData2.agg(skewness('a)),
+      Row(0.0))
+
+    checkAnswer(
```

Comment:
@holdenk @JihongMA Thanks for the suggestions! I updated `SQLQuerySuite` with a private `checkAnswerWithTol` helper function that solves the problem for the specific tests written so far. However, there are tests in `DataFrameAggregateSuite` that could fail if the test data ever changes - they too could benefit from a `checkAnswer` function that accepts a tolerance. I'm not sure if we can leave as is for now or perhaps the helper function should be added to `QueryTest` object so that it can be used in both test suites. Open to suggestions...